tensorboard/webapp/metrics/views/card_renderer/image_card_container.ts

```diff
@@ -299,6 +276,43 @@ export class ImageCardContainer implements CardRenderer, OnInit, OnDestroy {
         return selectedStepsInRange;
       })
     );
+
+    this.stepIndex$ = combineLatest(
+      this.store.select(getCardStepIndex, this.cardId),
+      this.selectedTime$,
+      this.selectedSteps$,
+      this.stepValues$
+    ).pipe(
+      map(([stepIndex, selectedTime, selectedSteps, stepValues]) => {
+        if (!selectedTime || selectedTime.endStep) return stepIndex;
+
+        const nextStepInex = stepValues.indexOf(selectedSteps[0]);
+        if (nextStepInex === -1) return stepIndex;
+        return nextStepInex;
+      })
+    );
+
+    const timeSeriesAndStepIndex$ = combineLatest([
```

Comment:
All of this has not changed. Is there a reason you moved it all? I would rather not mess up the git blame history on these.

Reply:
That's a fair question. Variables can only be used after they're defined.  (or they become undefined) 
`timeSeriesAndStepIndex$` needs `stepIndex$` which now needs `selectedSteps$`(and others, but the critical one is selectedSteps$)
that's why `stepIndex$` needed moved down and all other observables followed.

the chain is
selectedSteps$ -> stepIndex$ -> timeSeriesAndStepIndex$ -> stepDatum$ -> imageUrl$

Reply:
I see. Well I wish it were not the case but, so be it.